core/src/main/scala/chisel3/RawModule.scala

```diff
@@ -253,6 +293,11 @@ abstract class RawModule extends BaseModule {
     Builder.currentBlock.get.addSecretCommand(rhs)
   }
 
+  protected[chisel3] override def moduleBuilt(): Unit = {
+    // Evaluate any afterModuleBuilt generators.
+    _afterModuleBuilt.foreach { gen => gen() }
+  }
```

Comment:
Nit:
```suggestion
  // Evaluate any afterModuleBuilt generators.
  protected[chisel3] override def moduleBuilt(): Unit = _afterModuleBuilt.foreach(_())
```

If that is too weird, then `_.apply()` may be better.